### PR TITLE
Replace logger, add more logging options including files

### PR DESCRIPTION
**Pull request by notgne2**
This replaces the logger which has both value in aesthetics, but also gives us file logging, which can be useful in debugging #38 and #32

EDIT: Updates to logging may have even _fixed_ #38 and #32

### CODE REVIEW

File: src/utils/mod.rs
Comment (balsoft) [inline src/utils/mod.rs:64]: Eww

Comment (notgne2) [reply src/utils/mod.rs:64]: It won't take a closure :/

Comment (balsoft) [reply src/utils/mod.rs:64]: I mean, at least pull out that `match`

Comment (notgne2) [reply src/utils/mod.rs]: Done

Comment (mkaito) [reply src/utils/mod.rs]: Please no

File: src/utils/mod.rs
Comment (antifuchs) [inline src/utils/mod.rs:50]: I don't know if flexi_logger supports fields or extended (per-process) metadata, but it would be good to get the PID of the process writing the log line here, to help me figure out https://github.com/serokell/deploy-rs/issues/38#issuecomment-753011354 (:

Comment (notgne2) [reply src/utils/mod.rs:50]: That issue shouldn't exist anymore in this PR

**Comment by balsoft** [comment]
Now magic rollback doesn't confirm that we're still able to authenticate via SSH, merely that the server is still reachable.

**Comment by notgne2** [comment]
@balsoft I don't see how that's true, am I missing something here? The confirmation process works the same way, but instead of `deploy` waiting for the `activate` ssh connection to end before confirming, it waits for the new `wait` ssh connection to end (which occurs when the confirmation file is _created_), and instead of `activate` forking, it just ignores `SIGHUP`

**Comment by balsoft** [comment]
I thought the second connection is started before the activation actually occurs, am I wrong?

**Comment by notgne2** [comment]
- `ssh example.com activate &` (runs in the background, we don't care about it anymore)
- `ssh example.com wait` (waits for the confirmation file to exist, which happens after activation)
- `ssh example.com rm /tmp/confirmation-file` (actually confirm so the `activate` process from earlier doesn't rollback)

It _should_ be working like this, it only isn't if I have written it wrong or haven't thought about something correctly

**Comment by balsoft** [comment]
Hm, ok, I've re-read the code and it seems that the logic is indeed 
correct.

**Comment by notgne2** [comment]
I'll take this as a sign it's time for me to start documenting my code

**Comment by notgne2** [comment]
(on second thought, there's a lot of code cleanup and documentation to be done everywhere, so I think I'll wait until post-merge)

**Review by mkaito — APPROVED** [review]
Seems like a solid improvement.

**Review by balsoft — APPROVED** [review]
Looks good, but haven't actually tested it.

**Pull request merged**
